topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 GB`
+
+h|Kafka property name
+|`segment.bytes`
+|===
+--
+
+Segment time::
++
+--
+Segment time is the amount of time after which the current log segment is rolled even if the segment file is not full. This property ensures that the topic cleanup policy can delete or compact old messages as needed.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`1 week`
+
+h|Kafka property name
+|`segment.ms`
+|===
+--
+
+Segment jitter time::
++
+--
+Segment jitter time is the maximum delay for log segment rolling. This delay prevents bursts of log segment rolling activity.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`0 seconds`
+
+h|Kafka property name
+|`segment.jitter.ms`
+|===
+--
+
+File delete delay::
++
+--
+File delete delay is the amount of time that a file is retained in the system before the file is deleted.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`1 minute`
+
+h|Kafka property name
+|`file.delete.delay.ms`
+|===
+--
+
+Preallocate log segment files::
++
+--
+Preallocate log segment files determines whether to preallocate the file on disk when creating a new log segment. This property ensures sufficient disk space for log segments.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Boolean
+
+h|Default value
+|`Disabled`
+
+h|Kafka property name
+|`preallocate`
+|===
+--
+
+=== Index
+
+These properties control the indexing of the log.
+
+Index interval size::
++
+--
+Index interval size is the number of bytes between each index entry to its offset index. The default setting indexes a message about every 4096 bytes. More indexing enables reads to be closer to the exact position in the log but makes the index larger.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`4 KB`
+
+h|Kafka property name
+|`index.interval.bytes`
+|===
+--
+
+Segment index size::
++
+--
+Segment index size is the size of the index that maps offset to file positions.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`10 MB`
+
+h|Kafka property name
+|`segment.index.bytes`
+|===
+--
+
+=== Flush
+
+These properties control the frequency of the flushing of the log.
+
+Flush interval messages::
++
+--
+Flush interval messages is the number of messages between each data flush to the log.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`292271023 years, 2 weeks, 2 days, 13 hours, 12 minutes, 56.064 seconds`
```

Comment:
I think the Default value here is just a number representing the number of messages and not a duration

Reply:
@b1zzu Good point actually. But that's what it shows in the UI. See image below. Same default value for both the flush interval message and the flush interval time. So may be a bug?

![image](https://user-images.githubusercontent.com/29641047/119659992-bc178b80-bdfc-11eb-83db-5b254e4a6eb6.png)

Reply:
```json
{
      "key": "flush.messages",
      "value": "9223372036854775807"
},
{
      "key": "flush.ms",
      "value": "9223372036854775807"
}
```

It appears that the same value is returned from the Kafka Admin API. I'm not sure if that is correct, it could be worth logging an issue on https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api

Reply:
@craicoverflow ack. I've created the issue: https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/issues/65

Reply:
@b1zzu , @craicoverflow , @PaulRMellor, turns out to be an issue just in the UI, so I've submitted the UI bug here: https://github.com/bf2fc6cc711aee1a0c2a/kafka-ui/issues/128  I closed the old API bug that is now irrelevant. Meanwhile I've updated the docs accordingly. Will resolve this thread once the UI is fixed and I've confirmed for docs. Thanks.

Reply:
Fixed in PR https://github.com/bf2fc6cc711aee1a0c2a/kafka-ui/pull/108. Resolving.